lib/segment/src/index/struct_payload_index.rs

```diff
@@ -56,11 +56,18 @@ impl StructPayloadIndex {
     pub fn estimate_field_condition(
         &self,
         condition: &FieldCondition,
+        nested_path: Option<&JsonPathPayload>,
     ) -> Option<CardinalityEstimation> {
-        self.field_indexes.get(&condition.key).and_then(|indexes| {
+        let full_path = JsonPathPayload::extend_or_new(nested_path, &condition.key);
+        self.field_indexes.get(&full_path.path).and_then(|indexes| {
+            // rewrite condition with fullpath to enable cardinality estimation
+            let full_path_condition = FieldCondition {
+                key: full_path.path.clone(),
+                ..condition.clone()
```

Comment:
nit: is it possible without clone? (fine, if not)

Reply:
I don't think so; then we'll have to clone on a different level.